### DISCUSSION TITLE
[Enhancement] Adapt ".NET Multi-platform App UI" to the same licensing policy as .NET Core

### DISCUSSION THREAD

**Discussion by JensNordenbro** [dotnet/maui#73] [General]
## Summary
Please only accept using code and dependencies (direct or transitive) that is/are licensed under MIT and Apache as with the rest of .NET Core. 
This is to allow development of software that will not inherit licenses such as GPL, LGPL, *GPL so that ".NET Multi-platform App UI" can be used in a commercial context for all target platforms.

## API Changes
Remove code+dependencies that are not licensed under MIT and Apache. For example gtk# and gtk seems ~~to be GPL and LGPL~~and that will stop some commercial applications from targeting Linux.
Edit: Somebody mentioned that gtk* is actually licensed under GNU Library General Public License which is better however still it is creating more requirements on the consumer that is hard to cope with long term for a commercial entity. Also FWIW this license is not recommended by FSF and could be changed into L-GPL. 

Aside from the commercial aspect I as an OSS developer would like to be able to choose a LICENSE (for example MIT) FREELY as is possible with MIT for .NET Core. 

## Intended Use Case
As a developer using ".NET Multi-platform App UI" I would like to be able to create software where I can choose the LICENSE of that software while still being able to utilize the full software stack of ".NET Multi-platform App UI". 

Side note:  https://github.com/dotnet/core/blob/master/license-information-windows.md

**Comment by probonopd** [comment]
> Adapt MAUI

Please do not refer to it under this name, as MAUI is a registered trademark used for pre-existing open source projects.

According to https://github.com/dotnet/maui/issues/35#issuecomment-631056170, 

> the official legal name is .NET Multi-platform App UI 

Thanks for your kind consideration.

**Comment by Happypig375** [comment]
Can someone stop this Internet troll? He just disrupts constructive conversations.

**Comment by JensNordenbro** [comment]
I changed @probonopd , @Happypig375 ; the name does not matter to me, I can adapt :)

**Comment by R2D221** [comment]
This is not a legal document. You can say MAUI all you want.

**Comment by john-h-k** [comment]
I agree, maui loses a lot of its potential uses cases if it becomes copyleft restricted 

> Please do not refer to it under this name, as MAUI is a registered trademark used for pre-existing open source projects.

The MAUI trademark is owned by a german telecomms company from the look of it. So I'd advise you to start complaining to MauiKit that they are violating that registered trademark

**Comment by JensNordenbro** [comment]
haha guys, let´s not make this about the name :) 

Thanks @john-h-k for partially staying on topic :)

**Comment by BreyerW** [comment]
I also second that gtk# and any libraries that restrict usage even if they are technically open source (so any GPL and the like) should be jettisoned out of main repro and ideally provided as external nuget package making licensing and managing dependecies with such different licencies less of a hassle (its easier to add/remove just one nuget package than to remove whole dll then import source code then remove source code with offensive license)

Ideally all libraries in main repro should be mit apache or similar everything else as external nuget package and in future develop c# pure (or as close as possible to pure) binding to most popular linux windowing libraries (x11 and/or wayland?) i heard thats the route avalonia ui is going to take sooner or later to get rid of gpl-like licenses and trim down dependencies (whole gtk is useless baggage when u need only windowing and events like avalonia since avalonia provides its own ui)

**Comment by probonopd** [comment]
> The MAUI trademark is owned by a german telecomms company from the look of it. So I'd advise you to start complaining to MauiKit that they are violating that registered trademark

The German company is a main sponsor of the KDE open source project in case you are not aware, which might explain why KDE projects are using this name.

**Comment by bjornen77** [comment]
I agree that it is very important to use the same licensing as the rest of .NET Core.

**Comment by Conan-Kudo** [comment]
> Edit: Somebody mentioned that gtk* is actually licensed under GNU Library General Public License which is better however still it is creating more requirements on the consumer that is hard to cope with long term for a commercial entity. Also FWIW this license is not recommended by FSF and could be changed into L-GPL.

As someone who has actually helped with commercial compliance, if redistributing the source code of libraries you bundle is such a huge burden (the truth is, it's not), then you have significant other problems. Also, the GNU Library General Public License is identical to the GNU Lesser General Public License. The latter is a rename of the former.

**Comment by JensNordenbro** [comment]
Hi @Conan-Kudo! 
The problem is not to include the source code or provide the source code in the aftermath. 
The problem is that suddenly some parts of .NET Core requires compliance to L-GPL (or similar) licenses that require you to do this. 
This from a state where it did not originally! 
This will make companies that do not do extensive license compliance investigations & mitigations vulnerable to potential trolls who once contributed to the code and then uses lawyers to try to extract money out of companies. These guys do in fact exist! 

This new burden is completely unnecessary and it reduces MAUI:s appeal to me and I know many others.
  
Also staying away from MAUI might not help, since once the genie is out of the bottle new dependencies towards L-GPL can be argued to be taken inside more core parts, which would make the situation even worse.
